src/compiler/checker.ts

```diff
@@ -233,6 +233,60 @@ namespace ts {
                 return resolveName(location, escapeLeadingUnderscores(name), meaning, /*nameNotFoundMessage*/ undefined, /*nameArg*/ undefined);
             },
             getJsxNamespace: () => unescapeLeadingUnderscores(getJsxNamespace()),
+
+            isIdenticalTo: (a, b) => checkTypeRelatedTo(a, b, identityRelation, /*errorNode*/ undefined),
+            isSubtypeOf: (a, b) => checkTypeRelatedTo(a, b, subtypeRelation, /*errorNode*/ undefined),
+            isAssignableTo: (a, b) => checkTypeRelatedTo(a, b, assignableRelation, /*errorNode*/ undefined),
+            isComparableTo: areTypesComparable,
+            isInstantiationOf: (a, b) => {
+                return a && b && (a.target === b);
+            },
+
+            lookupGlobalType: name => {
+                const symbol = getSymbol(globals, escapeLeadingUnderscores(name), SymbolFlags.Type);
+                return symbol ? getDeclaredTypeOfSymbol(symbol) : unknownType;
+            },
+            lookupGlobalValueType: name => {
+                const symbol = getSymbol(globals, escapeLeadingUnderscores(name), SymbolFlags.Value);
+                return symbol ? getTypeOfSymbol(symbol) : unknownType;
+            },
+            lookupTypeAt: (name, node) => {

```

Comment:
wounder if we should just expose resolveName instead..

Reply:
Exposing it directly would make public way too many of the internal details, like usage checking, that are built into it. It's very overloaded, and the public API edition of its functionality _should_ be simplified, I think. Looking at this again, I think `Type/Value` may be better qualifiers on the function names than `Type/ValueType`, though.

Reply:
Should use `resolveName`